mautic/mautic
# refactoring & Test case for send email to dnd contact

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Refactoring & Test case for send email to dnd contact which was implemented in https://github.com/mautic/mautic/pull/11787/files



#### Steps to test this PR:



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send an individual email, an email via API /emails/ID/contact/CONTACT_ID/send, transactional email by form and by campaign.
3. Check the headers of the emails. There is no 'List-Unsubscribe'.
4. Send a marketing email by form and by campaign.
5. Check the headers of the emails. There is 'ListUnsubscribe'.
